CDCgov/prime-reportstream
# Refactor: One `Link` to rule them all

This PR implements a `USLink` component system, including specialty uses like `USExtLink`, to replace all rogue `a`, `Link` and `NavLink` uses with a consolidated application of styles and security. It updates all found links, though it has its limitations:

- We cannot use relative links within React Router from markdown pages (fixed if we add `MDX` support).
- We cannot use relative links within sections of `Home` that use `dangerouslySetInnerHTML` (though, I'd argue we shouldn't be using this method _at all_).

Test Steps:
1. Search the Repository for uses of `a`, `Link`, and `NavLink`. Your results should show the only uses of those tags within the `USLink.tsx` file
2. Ensure that all Navigation links are working in our header, including pages, the org selector, and "log out"
3. Ensure all links are working from public site pages (See known issues for details)
4. Ensure all links are working from Daily Data, Submission History, Value Sets, Last Mile Failures, and Receiver Status Dashboard tables
5. Ensure all links are working within the Admin Organizations table (the admin landing page)

>**NOTE:** Some links are wrapping `button` elements; ensure you are testing buttons that indicate navigation to a new component, such as "Edit" on the Admin Organizations table, or "New" in the organization settings UI.

## Changes
- Updates uses of `a`, `Link`, and `NavLink` to `USLink` and its specializations.
- Adds "Skip Nav" link 

Site link
<img width="191" alt="Screen Shot 2022-12-29 at 4 52 37 PM" src="https://user-images.githubusercontent.com/4022304/210014580-7a7fbad9-d1f6-4b3d-b2c3-994c100a9873.png">

External link
<img width="183" alt="Screen Shot 2022-12-29 at 4 50 33 PM" src="https://user-images.githubusercontent.com/4022304/210014495-894649c5-43ee-4286-9891-2e2409a70aee.png">

Navigation link w/ active style
<img width="86" alt="Screen Shot 2022-12-29 at 4 50 45 PM" src="https://user-images.githubusercontent.com/4022304/210014496-e1333cbb-6182-4e9f-91c4-0d2f996c585a.png">

Special case where buttons are links:
<img width="246" alt="Screen Shot 2022-12-29 at 4 49 38 PM" src="https://user-images.githubusercontent.com/4022304/210014494-a77a568b-afe1-4b76-8131-96d65a67030a.png">


## Checklist

### Testing

### Process

## Linked Issues
- Fixes #7730 
- Fixes #2772 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
